src/MPAGSCipher/CaesarCipher.cpp

```diff
@@ -1,38 +1,49 @@
-#include "RunCaesarCipher.hpp"
+#include "CaesarCipher.hpp"
+#include "CipherMode.hpp"
 
+#include <iostream>
 #include <string>
-#include <vector>
 
-std::string runCaesarCipher(const std::string& inputText, const std::size_t key,
-                            const bool encrypt)
-{
+CaesarCipher::CaesarCipher(const std::size_t cipherKey) {
+    key_ = cipherKey;
+}
+CaesarCipher::CaesarCipher(const std::string cipherKey) {
+    if (cipherKey.empty()) {
+        key_ = 0;
+    }
+    else {
+        key_ = std::stoul(cipherKey);
+    }
+}
+
+std::string CaesarCipher::applyCipher(const std::string inputText, const CipherMode mode) {
```

Comment:
This member function should be declared `const` because it does not modify the state of the object:
```suggestion
std::string CaesarCipher::applyCipher(const std::string inputText, const CipherMode mode) const {
```